### PR TITLE
[BugFix]Making String.base64Decoded a safe base64 decode by ignore non-base64 characters

### LINKED ISSUE #956: decode base64 string fail
string as under can decode at https://base64.us/, 
but can't decode string's base64Decoded property

IyBUcnVzdCBXYWxsZXQgQXNzZXRzIEluZm8KCiFbQ2hlY2tdKGh0dHBzOi8v\nZ2l0aHViLmNvbS90cnVzdHdhbGxldC9hc3NldHMvd29ya2Zsb3dzL0NoZWNr\nL2JhZGdlLnN2ZykKCiMjIE92ZXJ2aWV3ClRydXN0IFdhbGxldCB0b2tlbiBy\nZXBvc2l0b3J5IGlzIGEgY29tcHJlaGVuc2l2ZSwgdXAtdG8tZGF0ZSBjb2xs\nZWN0aW9uIG9mIGluZm9ybWF0aW9uIGFib3V0IHNldmVyYWwgdGhvdXNhbmRz\nICghKSBvZiBjcnlwdG8gdG9rZW5zLgoKW1RydXN0IFdhbGxldF0oaHR0cHM6\nLy90cnVzdHdhbGxldC5jb20pIHVzZXMgdG9rZW4gbG9nb3MgZnJvbSB0aGlz\nIHNvdXJjZSwgYWxvbmdzaWRlIGEgbnVtYmVyIG9mIG90aGVyIHByb2plY3Rz\nLgoKVGhlIHJlcG9zaXRvcnkgY29udGFpbnMgdG9rZW4gaW5mbyBmcm9tIHNl\ndmVyYWwgYmxvY2tjaGFpbnMsIGluZm8gb24gZEFwcHMsIHN0YWtpbmcgdmFs\naWRhdG9ycywgZXRjLgpGb3IgZXZlcnkgdG9rZW4gYSBsb2dvIGFuZCBvcHRp\nb25hbCBhZGRpdGlvbmFsIGluZm9ybWF0aW9uIGlzIGF2YWlsYWJsZSAoc3Vj\naCBkYXRhIGlzIG5vdCBhdmFpbGFibGUgb24tY2hhaW4pLgoKU3VjaCBhIGxh\ncmdlIGNvbGxlY3Rpb24gY2FuIGJlIG1haW50YWluZWQgb25seSB0aHJvdWdo\nIGEgY29tbXVuaXR5IGVmZm9ydCwgc28gX2ZlZWwgZnJlZSB0byBhZGQgeW91\nciB0b2tlbl8uCgo8Y2VudGVyPjxpbWcgc3JjPSdodHRwczovL3RydXN0d2Fs\nbGV0LmNvbS9hc3NldHMvaW1hZ2VzL21lZGlhL2Fzc2V0cy9ob3Jpem9udGFs\nX2JsdWUucG5nJyBoZWlnaHQ9IjIwMCI+PC9jZW50ZXI+CgojIyBIb3cgdG8g\nYWRkIHRva2VuCgpQbGVhc2Ugbm90ZSB0aGF0IF9fYnJhbmQgbmV3IHRva2Vu\ncyBhcmUgbm90IGFjY2VwdGVkX18sCnRoZSBwcm9qZWN0cyBoYXZlIHRvIGJl\nIHNvdW5kLCB3aXRoIGluZm9ybWF0aW9uIGF2YWlsYWJsZSwgYW5kIF9fbm9u\nLW1pbmltYWwgY2lyY3VsYXRpb25fXwooZm9yIGxpbWl0IGRldGFpbHMgc2Vl\nIGh0dHBzOi8vY29tbXVuaXR5LnRydXN0d2FsbGV0LmNvbS90L2hvdy10by1z\ndWJtaXQtYS10b2tlbi1sb2dvLzM4NjMpLgoKIyMjIEFzc2V0cyBBcHAKClRo\nZSBbQXNzZXRzIHdlYiBhcHBdKGh0dHBzOi8vYXNzZXRzLnRydXN0d2FsbGV0\nLmNvbSkgY2FuIGJlIHVzZWQgZm9yIG1vc3QgbmV3IHRva2VuIGFkZGl0aW9u\ncyAoR2l0aHViIGFjY291bnQgaXMgbmVlZGVkKS4KCiMjIyBRdWljayBzdGFy\ndGVyCgpEZXRhaWxzIG9mIHRoZSByZXBvc2l0b3J5IHN0cnVjdHVyZSBhbmQg\nY29udHJpYnV0aW9uIGd1aWRlbGluZXMgYXJlIGxpc3RlZCBvbiB0aGUKW0Rl\ndmVsb3BlcnMgc2l0ZV0oaHR0cHM6Ly9kZXZlbG9wZXIudHJ1c3R3YWxsZXQu\nY29tL2FkZF9uZXdfYXNzZXQpLgpIZXJlIGlzIGEgcXVpY2sgc3RhcnRlciBz\ndW1tYXJ5IGZvciB0aGUgbW9zdCBjb21tb24gdXNlIGNhc2UuCgoqKkFkZGlu\nZyBhbiBFUkMyMCB0b2tlbiBjaGVja2xpc3QqKjoKLSBbIF0gTWFrZSBzdXJl\nIHlvdXIgc21hcnRjb250cmFjdCBoYXMgbW9yZSB0aGFuIDIsNTAwIGFkZHJl\nc3MgaG9sZGVycywgb3RoZXJ3aXNlIHlvdSB3aWxsIGJlIHJlamVjdGVkCi0g\nWyBdIEZvcmsgdGhlIEdpdGh1YiByZXBvc2l0b3J5Ci0gWyBdIENyZWF0ZSBm\nb2xkZXIgd2l0aCBuYW1lIG9mIHRva2VuIHNtYXJ0Y29udGFjdCBhZGRyZXNz\nIGluIFtfY2hlY2tzdW0gZm9ybWF0X10oaHR0cHM6Ly9kZXZlbG9wZXIudHJ1\nc3R3YWxsZXQuY29tL2FkZF9uZXdfYXNzZXQjY2hlY2tzdW1fZm9ybWF0KSBg\nYmxvY2tjaGFpbnMvZXRoZXJldW0vYXNzZXRzLzx0b2tlbl9zbWFydGNvbnRy\nYWN0X2FkZHJlc3M+L2AuCi0gWyBdIFRlbGwgeW91ciBkZXNpZ25lciB0aGF0\nIHRva2VuIGltYWdlIG11c3QgYmUgaW4gUE5HIGZvcm1hdCwgYXZvaWQgdHJh\nbnNwYXJlbnQgYmFja2dyb3VuZCwgcmVjb21tZW5kZWQgc2l6ZSAyNTZ4MjU2\ncHgsIG1heC4gNTEyeDUxMnB4LCB3aXRoIG1heCBmaWxlIHNpemUgb2YgMTAw\na0IsIGZvciBmdXJ0aGVyIGRldGFpbHMgcmVhZCBbaW1hZ2UgcnVsZXNdKGh0\ndHBzOi8vZGV2ZWxvcGVyLnRydXN0d2FsbGV0LmNvbS9hZGRfbmV3X2Fzc2V0\nI2ltYWdlLXJlcXVpcmVtZW50cykuCi0gWyBdIFVwbG9hZCB5b3VyIGxvZ28g\nd2l0aCBmaWxlIG5hbWVkIGBsb2dvLnBuZ2AgdG8gcHJldmlvdXNseSBjcmVh\ndGVkIGZvbGRlciB3aXRoIHNtYXJ0Y29udHJhY3QgYWRkcmVzcywgYW5kIGlm\nIHlvdSBkb25lIGFsbCBjb3JyZWN0bHkgeW91ciBwYXRoIHNob3VsZCBsb29r\nIGxpa2UgdGhpcy4gYGJsb2NrY2hhaW5zL2V0aGVyZXVtL2Fzc2V0cy8weDEy\nMzQ1Njc0NjFkM2Y4RGI3NDk2NTgxNzc0QmQ4NjlDODNENTFjOTMvbG9nby5w\nbmdgCi0gWyBdIENyZWF0ZSBgaW5mby5qc29uYCBmaWxlIHdpdGggaW5mbyBh\nYm91dCB0aGUgdG9rZW4vcHJvamVjdAotIFsgXSBDcmVhdGUgYSBwdWxsIHJl\ncXVlc3QgdG8gdGhlIG1haW4gcmVwbwotIFsgXSBQYXkgdGhlIHByb2Nlc3Np\nbmcgZmVlCgojIyBEb2N1bWVudGF0aW9uCgpGb3IgZGV0YWlscywgc2VlIHRo\nZSBbRGV2ZWxvcGVycyBzaXRlXShodHRwczovL2RldmVsb3Blci50cnVzdHdh\nbGxldC5jb20vYWRkX25ld19hc3NldCk6CgotIFtDb250cmlidXRpb24gZ3Vp\nZGVsaW5lc10oaHR0cHM6Ly9kZXZlbG9wZXIudHJ1c3R3YWxsZXQuY29tL2Fk\nZF9uZXdfYXNzZXQjY29udHJpYnV0aW9uLWd1aWRlbGluZXMpCgotIFtSZXBv\nc2l0b3J5IGRldGFpbHNdKGh0dHBzOi8vZGV2ZWxvcGVyLnRydXN0d2FsbGV0\nLmNvbS9hZGRfbmV3X2Fzc2V0I3JlcG9zaXRvcnktZGV0YWlscykKCiMjIFNj\ncmlwdHMKClRoZXJlIGFyZSBzZXZlcmFsIHNjcmlwdHMgYXZhaWxhYmxlIGZv\nciBtYWludGFpbmVyczoKCi0gYG5wbSBydW4gY2hlY2tgIC0tIEV4ZWN1dGUg\ndmFsaWRhdGlvbiBjaGVja3M7IGFsc28gdXNlZCBpbiBjb250aW51b3VzIGlu\ndGVncmF0aW9uLgotIGBucG0gcnVuIGNoZWNrLXNhbml0eWAgLS0gU3RyaWN0\nIHN1YnNldCBvZiBjaGVja3MKLSBgbnBtIHJ1biBmaXhgIC0tIFBlcmZvcm0g\nYXV0b21hdGljIGZpeGVzIHdoZXJlIHBvc3NpYmxlCi0gYG5wbSBydW4gZml4\nLXNhbml0eWAgLS0gU3RyaWN0ZXIgc3Vic2V0Ci0gYG5wbSBydW4gdXBkYXRl\nQXV0b2AgLS0gUnVuIGF1dG9tYXRpYyB1cGRhdGVzIGZyb20gZXh0ZXJuYWwg\nc291cmNlcywgZXhlY3V0ZWQgcmVndWxhcmx5IChHaXRIdWIgYWN0aW9uKQot\nIGBucG0gcnVuIHVwZGF0ZWAgLS0gUnVuIG1hbnVhbCB1cGRhdGVzIGZyb20g\nZXh0ZXJuYWwgc291cmNlcywgZm9yIG1hbnVhbCB1c2UuCi0gYG5wbSB0ZXN0\nYCAtLSBSdW4gc2NyaXB0IHVuaXQgdGVzdHMKLSBgbnBtIGxpbnRgIC0tIFJ1\nbiBMaW50IHN0YXRpYyBjb2RlIGNoZWNrCgojIyBPbiBDaGVja3MKClRoaXMg\ncmVwbyBjb250YWlzIGEgc2V0IG9mIHNjcmlwdHMgZm9yIHZlcmlmaWNhdGlv\nbiBvZiBhbGwgdGhlIGluZm9ybWF0aW9uLiAgSW1wbGVtZW50ZWQgYXMgVHlw\nZXNjcmlwdCBzY3JpcHRzLCBhdmFpbGFibGUgdGhyb3VnaCBgbnBtIHJ1biBj\naGVja2AsIGFuZCBleGVjdXRlZCBpbiBDSSBidWlsZDsgY2hlY2tzIHRoZSB3\naG9sZSByZXBvLgpUaGVyZSBhcmUgc2ltaWxhciBjaGVjayBsb2dpYyBpbXBs\nZW1ldGVkOgotIGluIGFzc2V0cy1tYW5hZ2VtZW50IGFwcDsgZm9yIGNoZWNr\naW5nIGNoYW5nZWQgdG9rZW4gZmlsZXMgaW4gUFJzLCBvciB3aGVuIGNyZWF0\naW5nIGEgUFIuICBJbXBsZW1lbnRlZCBhcyBhIFR5cGVzY3JpcHQgbGlicmFy\neSwgY2hlY2tzIGRpZmZzLCBjYW4gYmUgcnVuIGZyb20gYnJvd3NlciBlbnZp\ncm9ubWVudC4KLSBpbiBtZXJnZS1mZWUtYm90LCB3aGljaCBydW5zIGFzIGEg\nR2l0SHViIGFwcCBzaG93cyByZXN1bHQgaW4gUFIgY29tbWVudC4gIEFsc28g\ndXNlcyBsaWJyYXJ5LCBidXQgZXhlY3V0ZXMgaW4gYSBub24tYnJvd3NlciBl\nbnZpcm9ubWVudC4KCiMjIFRyYWRpbmcgcGFpciBtYWludGVuYW5jZQoKSW5m\nbyBvbiBzdXBwb3J0ZWQgdHJhZGluZyBwYWlycyBhcmUgc3RvcmVkIGluIGB0\nb2tlbmxpc3QuanNvbmAgZmlsZXMuClRyYWRpbmcgcGFpcnMgY2FuIGJlIHVw\nZGF0ZWQgLS0KZnJvbSBVbmlzd2FwL0V0aGVyZXVtIGFuZCBQYW5ja2Fzd2Fw\nL1NtYXJ0Y2hhaW4gLS0gdXNpbmcgdXBkYXRlIHNjcmlwdCAoYW5kIGNoZWNr\naW5nIGluIGNoYW5nZXMpLgpNaW5pbWFsIGxpbWl0IHZhbHVlcyBmb3IgdHJh\nZGluZyBwYWlyIGluY2x1c2lvbiBhcmUgc2V0IGluIHRoZSBgY29uZmlnLnRz\nYCBmaWxlLgpUaGVyZSBhcmUgYWxzbyBvcHRpb25zIGZvciBmb3JjZS1pbmNs\ndWRlIGFuZCBmb3JjZS1leGNsdWRlIGluIHRoZSBjb25maWcuCgojIyBEaXNj\nbGFpbWVyClRydXN0IFdhbGxldCB0ZWFtIGFsbG93cyBhbnlvbmUgdG8gc3Vi\nbWl0IG5ldyBhc3NldHMgdG8gdGhpcyByZXBvc2l0b3J5LiBIb3dldmVyLCB0\naGlzIGRvZXMgbm90IG1lYW4gdGhhdCB3ZSBhcmUgaW4gZGlyZWN0IHBhcnRu\nZXJzaGlwIHdpdGggYWxsIG9mIHRoZSBwcm9qZWN0cy4KClRydXN0IFdhbGxl\ndCB0ZWFtIHdpbGwgcmVqZWN0IHByb2plY3RzIHRoYXQgYXJlIGRlZW1lZCBh\ncyBzY2FtIG9yIGZyYXVkdWxlbnQgYWZ0ZXIgY2FyZWZ1bCByZXZpZXcuClRy\ndXN0IFdhbGxldCB0ZWFtIHJlc2VydmVzIHRoZSByaWdodCB0byBjaGFuZ2Ug\ndGhlIHRlcm1zIG9mIGFzc2V0IHN1Ym1pc3Npb25zIGF0IGFueSB0aW1lIGR1\nZSB0byBjaGFuZ2luZyBtYXJrZXQgY29uZGl0aW9ucywgcmlzayBvZiBmcmF1\nZCwgb3IgYW55IG90aGVyIGZhY3RvcnMgd2UgZGVlbSByZWxldmFudC4K\n

### PR DESCRIPTION
**Pull request by Jayxiang**
fix #956

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [ ] New extensions are written in Swift 5.0.
- [ ] New extensions support iOS 10.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+, or use `@available` if not.
- [ ] I have added tests for new extensions, and they passed.
- [ ] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [ ] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by guykogus** [comment]
Personally I find it strange that we're handling unknown characters in a straightforward base64 decoding function. This way you could have a completely corrupted string that still gets decoded. The extra padding that was added in #801 makes sense, but handling unknown characters seems out of scope for me with this one.
Thoughts @omaralbeik @LucianoPAlmeida ?

**Comment by LucianoPAlmeida** [comment]
> Personally I find it strange that we're handling unknown characters in a straightforward base64 decoding function. This way you could have a completely corrupted string that still gets decoded. The extra padding that was added in #801 makes sense, but handling unknown characters seems out of scope for me with this one.
> Thoughts @omaralbeik @LucianoPAlmeida ?

I think this is fine, mostly because the way it is implemented today we already accept `"SGVsbG\n8gV29ybGQh".base64Decoded` as valid so it may seems inconsistent to accept the unknown char in the middle but not at the end because an implementation detail on handling padding. So IMO we can get this in =]

File: Tests/SwiftStdlibTests/StringExtensionsTests.swift [hunk 1]
```diff
@@ -16,6 +16,7 @@ final class StringExtensionsTests: XCTestCase {
     func testBase64Decoded() {
         XCTAssertEqual("SGVsbG8gV29ybGQh".base64Decoded, helloWorld)
         XCTAssertEqual("http://example.com/xxx", "aHR0cDovL2V4YW1wbGUuY29tL3h4eA".base64Decoded)
+        XCTAssertEqual(helloWorld, "SGVsbG8gV29ybGQh\n".base64Decoded)
```
Comment (LucianoPAlmeida) [inline Tests/SwiftStdlibTests/StringExtensionsTests.swift:19]: Can you just add a test case for `"SGVsbG\n8gV29ybGQh".base64Decoded` (\n in the middle)

**Review by LucianoPAlmeida — COMMENTED** [review]
A nit test case and changelog entry and that would be good!
Thanks @Jayxiang

**Comment by Jayxiang** [comment]
Is the changelog still added below 5.3.0? @LucianoPAlmeida

**Comment by LucianoPAlmeida** [comment]
> Is the changelog still added below 5.3.0? @LucianoPAlmeida

Yeap =]

File: Tests/SwiftStdlibTests/StringExtensionsTests.swift [hunk 2]
```diff
@@ -16,6 +16,7 @@ final class StringExtensionsTests: XCTestCase {
     func testBase64Decoded() {
         XCTAssertEqual("SGVsbG8gV29ybGQh".base64Decoded, helloWorld)
         XCTAssertEqual("http://example.com/xxx", "aHR0cDovL2V4YW1wbGUuY29tL3h4eA".base64Decoded)
+        XCTAssertEqual(helloWorld, "SGVsbG\n8gV29ybGQh".base64Decoded)
```
Comment (LucianoPAlmeida) [inline Tests/SwiftStdlibTests/StringExtensionsTests.swift:19]: I meant to have both :)
```suggestion
        XCTAssertEqual(helloWorld, "SGVsbG\n8gV29ybGQh".base64Decoded)
        XCTAssertEqual(helloWorld, "SGVsbG8gV29ybGQh\n".base64Decoded)
```

**Review by LucianoPAlmeida — APPROVED** [review]
LGTM, Thanks @Jayxiang

**Pull request merged**